apps/ui/lib/components/Event/index.tsx

```diff
@@ -14,35 +14,78 @@ import Message from './Message';
 export { parseMessage };
 
 const EventWithActor: React.FunctionComponent<any> = (props) => {
-	const { card, user, onCardVisible, targetCard } = props;
+	const { card, user, onCardVisible, targetCard, isDirectMention } = props;
+	const mentionsUserArray = card.data.payload?.mentionsUser || [];
+	const directMessageMarkersArray = card.markers || [];
+	const isDirectMessagePing =
+		directMessageMarkersArray[0]?.split('+')[0] === user.slug || false;
+	const isDirect = mentionsUserArray.includes(user.slug) || isDirectMessagePing;
 	const typeBase = props.card.type.split('@')[0];
-	return (
-		<CardLoader<UserContract>
-			id={helpers.getActorIdFromCard(props.card)}
-			type="user"
-			withLinks={['is member of']}
-		>
-			{(author) => {
-				const actor = helpers.generateActorFromUserCard(author);
-				if (typeBase === UPDATE) {
-					return (
-						<Update
-							onCardVisible={onCardVisible}
-							card={card}
-							user={user}
-							actor={actor}
-						/>
-					);
-				}
-				if (typeBase === LINK) {
-					return (
-						<LinkedCard actor={actor} card={card} targetCard={targetCard} />
-					);
-				}
-				return <Message {...props} actor={actor} />;
-			}}
-		</CardLoader>
-	);
+	if (isDirectMention) {
```

Comment:
This is a bit confusing for me, you have `isDirectMention` and `isDirect` used in conditionals, but without comments its very hard to read what's going on and why.

Reply:
Also most of the contents in the `else` block is a copy of whats in the `if` block. This should be consolidated to reduce duplicate code.

Reply:
it's because there's 2 booleans to keep track, if isDirectMention then display the filtered cards from the isDirect check. Else render all @ group and @ user-slug

Reply:
I understand the general logic flow, what I'm asking is that you reduce duplicate code and at least add comments to explain what's happening.

Reply:
Having clearer variable names may help here as well, `onlyDirectMentions` boolean instead of `isDirectMention` for example.